### PR TITLE
Refactor exceptions package

### PR DESCRIPTION
**Pull request by serious6**
* refactor exceptions package to new location under core.exception (with subpackages)
* deprecating some exceptions under core.exceptions.misc because they have an equivalent representative with `java.lang.Exception`
* small improvements to ArgumentException
   * changed inheritance (now `IllegalArgumentException`)
   * refactored constructors

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@serious6__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java [hunk 1]
```diff
@@ -573,7 +573,7 @@ public static String formatLogMessage(String entryKind, String logEntry)
    *
    * @param response the response
    * @return the string
-   * @throws microsoft.exchange.webservices.data.exception.EWSHttpException the eWS http exception
+   * @throws EWSHttpException the eWS http exception
```
Comment (vboctor) [inline src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java]: EWS

File: src/main/java/microsoft/exchange/webservices/data/core/exception/service/remote/CreateAttachmentException.java [hunk 2]
```diff
@@ -31,8 +32,7 @@
  * Represents an error that occurs when a call to the CreateAttachment web
  * method fails.
  */
-public final class CreateAttachmentException extends
-    ServiceRemoteException {// extends
+public final class CreateAttachmentException extends ServiceRemoteException {// extends
```
Comment (vboctor) [inline src/main/java/microsoft/exchange/webservices/data/core/exception/service/remote/CreateAttachmentException.java]: Any reason to keep the comment with 0xDEADCODE ?

File: src/main/java/microsoft/exchange/webservices/data/core/exception/service/remote/DeleteAttachmentException.java [hunk 3]
```diff
@@ -31,8 +32,7 @@
  * Represents an error that occurs when a call to the DeleteAttachment web
  * method fails.
  */
-public final class DeleteAttachmentException extends
-    ServiceRemoteException {// extends
+public final class DeleteAttachmentException extends ServiceRemoteException {// extends
```
Comment (vboctor) [inline src/main/java/microsoft/exchange/webservices/data/core/exception/service/remote/DeleteAttachmentException.java]: 0xDEADCODE

**Comment by vboctor** [comment]
:+1:

File: src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java (hunk 1 above)
Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/core/EwsUtilities.java]: fixed. Also fixed all other occurrences of this typo.

File: src/main/java/microsoft/exchange/webservices/data/core/exception/service/remote/CreateAttachmentException.java (hunk 2 above)
Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/core/exception/service/remote/CreateAttachmentException.java]: fixed

File: src/main/java/microsoft/exchange/webservices/data/core/exception/service/remote/DeleteAttachmentException.java (hunk 3 above)
Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/core/exception/service/remote/DeleteAttachmentException.java]: fixed

**Pull request merged**